internal/controller/nimcache_controller.go

```diff
@@ -119,31 +122,43 @@ func NewNIMCacheReconciler(client client.Client, scheme *runtime.Scheme, log log
 // - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.18.2/pkg/reconcile
 func (r *NIMCacheReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
 	logger := log.FromContext(ctx)
+	var err error
+	var result reconcile.Result
 
 	// Fetch the NIMCache instance
 	nimCache := &appsv1alpha1.NIMCache{}
-	if err := r.Get(ctx, req.NamespacedName, nimCache); err != nil {
+	if err = r.Get(ctx, req.NamespacedName, nimCache); err != nil {
 		if client.IgnoreNotFound(err) != nil {
 			logger.Error(err, "unable to fetch NIMCache", "name", req.NamespacedName)
 		}
 		return ctrl.Result{}, client.IgnoreNotFound(err)
 	}
 	logger.Info("Reconciling", "NIMCache", nimCache.Name)
+	previousStatusState := nimCache.Status.State
 
+	defer func() {
+		if err != nil {
+			r.GetEventRecorder().Eventf(nimCache, corev1.EventTypeWarning, appsv1alpha1.NimCacheConditionReconcileFailed,
+				"NIMCache %s in namespace %s reconcile failed, msg: %s", nimCache.Name, nimCache.Namespace, err.Error())
+		} else if previousStatusState != nimCache.Status.State {
+			r.GetEventRecorder().Eventf(nimCache, corev1.EventTypeNormal, nimCache.Status.State,
+				"NIMCache %s in namespace %s reconcile success, new state: %s", nimCache.Name, nimCache.Namespace, nimCache.Status.State)
```

Comment:
we can omit logging namespace as these are all namespace scoped events. Events already have a `Namespace` field associated with them.

Reply:
```$ kubectl  get events -A
NAMESPACE      LAST SEEN   TYPE      REASON              OBJECT                                                MESSAGE
nim-service    14m         Normal    Ready               nimpipeline/my-nim-pipelin1                           nimPipeline my-nim-pipelin1 in namespace nim-service status Ready, service states map[embedding:Ready reranking:Ready]
nim-service    14m         Normal    Ready               nimservice/reranking                                  NIMService reranking in namespace nim-service ready, msg: deployment "reranking" successfully rolled out